pages/api/cards.ts

```diff
@@ -0,0 +1,47 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { PrismaClient } from "@prisma/client";
+import { Card } from "@prisma/client";
+const prisma = new PrismaClient();
+
+export default async (req: NextApiRequest, res: NextApiResponse<Card>) => {
+  if (req.method === "POST") {
+    const newCard = JSON.parse(req.body);
+    const saveCard = await prisma.card.upsert({
+      where: {
+        id: newCard.selectedCard?.id || "-1",
+      },
+      update: {
+        answer: newCard.answer,
+        question: newCard.question,
+      },
+      create: {
+        question: newCard.question,
+        answer: newCard.answer,
+        nextReviewDate: newCard.selectedCard.nextReviewDate,
+        deckId: newCard.selectedCard.deckId,
+        level: newCard.selectedCard.level,
```

Comment:
maybe you could deconstruct newCard here

Reply:
Done!